client/my-sites/stats/feedback/index.tsx

```diff
@@ -90,16 +97,98 @@ function FeedbackCard( { clickHandler }: FeedbackPropsInternal ) {
 	);
 }
 
+function storePanelDismissDate() {
+	localStorage.setItem( FEEDBACK_KEY_DISMISS_DATE, new Date().toISOString() );
+}
+
+function storeFeedbackSubmissionDate() {
+	localStorage.setItem( FEEDBACK_KEY_SUBMISSION_DATE, new Date().toISOString() );
+}
+
+function getDiffInMilliseconds( isoDate: string ) {
+	return new Date().getTime() - new Date( isoDate ).getTime();
+}
+
+function getDaysFromMilliseconds( milliseconds: number ) {
+	return Math.floor( milliseconds / ( 1000 * 60 * 60 * 24 ) );
+}
+
+function getMonthsFromMilliseconds( milliseconds: number ) {
+	return Math.floor( milliseconds / ( 1000 * 60 * 60 * 24 * 30 ) );
+}
+
+function getDaysFromISODate( isoDate: string ) {
+	return getDaysFromMilliseconds( getDiffInMilliseconds( isoDate ) );
+}
+
+function getMonthsFromISODate( isoDate: string ) {
+	return getMonthsFromMilliseconds( getDiffInMilliseconds( isoDate ) );
+}
+
+function didSubmitFeedbackWithinSixMonths() {
+	const submissionDate = localStorage.getItem( FEEDBACK_KEY_SUBMISSION_DATE );
+	if ( ! submissionDate ) {
+		return false;
+	}
+
+	if ( getMonthsFromISODate( submissionDate ) <= 6 ) {
+		return true;
+	}
+
+	return false;
+}
+
+function didDismissPanelWithinSevenDays() {
+	const dismissDate = localStorage.getItem( FEEDBACK_KEY_DISMISS_DATE );
+	if ( ! dismissDate ) {
+		return false;
+	}
+
+	if ( getDaysFromISODate( dismissDate ) <= 7 ) {
+		return true;
+	}
+
+	return false;
+}
+
+function shouldShowFloatingPanel() {
+	if ( didSubmitFeedbackWithinSixMonths() ) {
+		return false;
+	}
+
+	if ( didDismissPanelWithinSevenDays() ) {
+		return false;
+	}
+
+	return true;
+}
+
 function StatsFeedbackController( { siteId }: FeedbackProps ) {
 	const [ isOpen, setIsOpen ] = useState( false );
-	const [ isFloatingPanelOpen, setIsFloatingPanelOpen ] = useState( true );
+	const [ isFloatingPanelOpen, setIsFloatingPanelOpen ] = useState( false );
+
+	useEffect( () => {
+		if ( shouldShowFloatingPanel() ) {
+			setTimeout( () => {
+				setIsFloatingPanelOpen( true );
+			}, FEEDBACK_PANEL_PRESENTATION_DELAY );
+		}
+	}, [] );
 
 	const handleButtonClick = ( action: string ) => {
 		if ( action === FEEDBACK_ACTION_SEND_FEEDBACK ) {
 			setIsOpen( true );
 		}
 		if ( action === FEEDBACK_ACTION_DISMISS_FLOATING_PANEL ) {
+			storePanelDismissDate();
```

Comment:
we should probably just re-use the notice API for the purpose. On notice dismiss, we simply pass in how long they should stay hidden. Example could found here:

https://github.com/Automattic/wp-calypso/blob/7fec0e316c61e822b1c434906370c0c8c3528895/client/my-sites/stats/stats-notices/do-you-love-jetpack-stats-notice.tsx#L47

Reply:
Yeah. Once that PR lands, I can update the logic in the `StatsFeedbackController` component to use the updated hook.

Reply:
Sorry, once the DIFF lands. — D160350-code

Reply:
The Diff mentioned above has been deployed. @a8ck3n

Reply:
Thanks for the ping @dognose24. I'll look at updating the display logic today. 👍